src/systems/si-imperial/include/units/isq/si/imperial/thermodynamic_temperature.h

```diff
@@ -0,0 +1,120 @@
+// The MIT License (MIT)
+//
+// Copyright (c) 2020 Yves Delley
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in all
+// copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+// SOFTWARE.
+
+#pragma once
+
+// IWYU pragma: begin_exports
+#include <units/isq/dimensions/thermodynamic_temperature.h>
+#include <units/quantity.h>
+#include <units/quantity_point.h>
+#include <units/reference.h>
+#include <units/symbol_text.h>
+// IWYU pragma: end_exports
+
+
+namespace units::isq::si::imperial {
+
+struct degree_fahrenheit : named_scaled_unit<degree_fahrenheit, basic_symbol_text{"\u00b0F", "`F"}, no_prefix, ratio(5, 9), kelvin> {};
+
+namespace detail {
+
+struct zero_fahrenheit_abs_temperature_ : scaled_unit<ratio(27315 * 9 - 5 * 3200, 900), kelvin> {};
+
+}  // namespace detail
+
+struct fahrenheit_temperature_origin : point_origin<dim_thermodynamic_temperature> {
+  using reference_point_origin = kelvin_temperature_origin;
+  static constexpr auto offset_to_reference =
+      si::thermodynamic_temperature<imperial::detail::zero_fahrenheit_abs_temperature_, int>(1);
+};
+
+template<UnitOf<dim_thermodynamic_temperature> U = degree_fahrenheit, Representation Rep = double>
+using fahrenheit_temperature_point = quantity_point<fahrenheit_temperature_origin, U, Rep>;
+
+} // units::isq::si::imperial
+
+namespace units::detail {
+
+template <> struct customary_origin_spec_for_unit<units::isq::si::imperial::degree_fahrenheit> {
+  using type = units::isq::si::imperial::fahrenheit_temperature_origin;
+};
```

Comment:
This manually separate specification seems to go against the design of the library.